src/states/elder/mod.rs

```diff
@@ -1325,19 +1261,39 @@ impl Elder {
     // Signs and proves the given `RoutingMessage` and wraps it in `SignedRoutingMessage`.
     fn to_accumulating_message(
         &self,
-        msg: RoutingMessage,
+        dst: DstLocation,
+        variant: Variant,
         node_knowledge_override: Option<u64>,
-    ) -> Result<AccumulatingMessage, RoutingError> {
-        let proof = self.chain.prove(&msg.dst, node_knowledge_override);
+    ) -> Result<AccumulatingMessage> {
+        let proof = self.chain.prove(&dst, node_knowledge_override);
         let pk_set = self.our_section_bls_keys().clone();
         let secret_key = self.chain.our_section_bls_secret_key_share()?;
 
-        AccumulatingMessage::new(msg, secret_key, pk_set, proof)
+        AccumulatingMessage::new(*self.our_prefix(), dst, variant, secret_key, pk_set, proof)
     }
 
     pub fn in_src_location(&self, src: &SrcLocation) -> bool {
         self.chain.in_src_location(src)
     }
+
+    // Verifies message but doesn't log anything on failure, only returns result.
+    fn verify_message_quiet(&self, msg: &Message) -> Result<bool, RoutingError> {
+        match msg.verify(self.chain.get_their_key_infos()) {
+            Ok(VerifyStatus::Full) => Ok(true),
+            Ok(VerifyStatus::ProofTooNew) if msg.dst.is_multiple() => {
+                // Proof is too new which can only happen if we've been already demoted but are
+                // lagging behind (or the sender is faulty/malicious). We can't handle the
+                // message ourselves but the other elders likely can.
+                Ok(false)
+            }
+            Ok(VerifyStatus::ProofTooNew) => Err(RoutingError::UntrustedMessage),
+            Err(error) => Err(error),
+        }
+    }
+
+    fn is_message_verified(&self, msg: &Message) -> bool {
```

Comment:
Not sure the name is sensible as it seem to suggest a cheap operation. (one already done).